### PR TITLE
Core: update start_blueos_core to add memory limit and logging

### LINKED ISSUE #2218: Services should be automatically restarted when killed due to high memory usage, and the user should be notified that this happened

### PR DESCRIPTION
**Pull request by Williangalvani**
(opening as draft because I have issues with shellcheck on my machine)

fix #2218

### CODE REVIEW

**Comment by rafaellehmkuhl** [comment]
Major services are being restarted on boot already. That includes the ArdupilotManager or mavlink2rest.

image

File: core/run-service.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+service_command=$1
+memory_limit_mb=$2
```
Comment (rafaellehmkuhl) [inline core/run-service.sh:4]: I think you're limiting services to 2MB 👀

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

File: core/run-service.sh (hunk 1 above)
Comment (Williangalvani) [reply core/run-service.sh:4]: NO U!

Comment (Williangalvani) [reply core/run-service.sh:4]: I added a debug line to be sure of what is going on

File: core/start-blueos-core [hunk 2]
```diff
@@ -69,31 +69,31 @@ fi
...
-    'bridget',"nice -19 $SERVICES_PATH/bridget/main.py",200
-    'commander',"$SERVICES_PATH/commander/main.py",200
-    'nmea_injector',"nice -19 $SERVICES_PATH/nmea_injector/nmea_injector/main.py",200
-    'helper',"$SERVICES_PATH/helper/main.py",200
-    'iperf3'," iperf3 --server --port 5201",200
-    'linux2rest',"linux2rest",200
-    'filebrowser',"nice -19 filebrowser --database /etc/filebrowser/filebrowser.db --baseurl /file-browser",200
-    'versionchooser',"$SERVICES_PATH/versionchooser/main.py",200
-    'pardal',"nice -19 $SERVICES_PATH/pardal/main.py",200
-    'ping',"nice -19 $SERVICES_PATH/ping/main.py",200
```
Comment (patrickelectric) [inline core/start-blueos-core]: you should put 200 inside a string probably

File: core/start-blueos-core [hunk 3]
```diff
@@ -71,7 +71,7 @@ fi
 PRIORITY_SERVICES=(
     'autopilot',250,"nice --19 $SERVICES_PATH/ardupilot_manager/main.py"
     'cable_guy',250,"$SERVICES_PATH/cable_guy/main.py"
-    'video',250,"nice --19 mavlink-camera-manager --default-settings BlueROVUDP --mavlink tcpout:127.0.0.1:5777 --gst-feature-rank omxh264enc=0,v4l2h264enc=250,x264enc=260 --log-path /var/logs/blueos/services/mavlink-camera-manager --verbose"
+    'video',1000,"nice --19 mavlink-camera-manager --default-settings BlueROVUDP --mavlink tcpout:127.0.0.1:5777 --gst-feature-rank omxh264enc=0,v4l2h264enc=250,x264enc=260 --log-path /var/logs/blueos/services/mavlink-camera-manager --verbose"
```
Comment (patrickelectric) [inline core/start-blueos-core:74]: meme-memes

Comment (Williangalvani) [reply core/start-blueos-core:74]: dont look at me :eyes:

Comment (patrickelectric) [reply core/start-blueos-core:74]: @joaoantoniocardoso is that expected with the leak ?

Comment (Williangalvani) [reply core/start-blueos-core:74]: in my tests mcm started at about 350mb, going then with light cockpit usage it stayed under 500mb. but idk how it behaves with multiple cameras and/or 4k streams

**Review by rafaellehmkuhl — APPROVED** [review]
Tested and it works.

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**